app/models/concerns/rule_remediation.rb

```diff
@@ -9,10 +9,13 @@ module RuleRemediation
   #  `.includes(profiles: :benchmark)`
   #
   def remediation_issue_id
-    # FIXME: Nondetermenistic canonical selection
-    profile = profiles.select(&:canonical?)&.first
-    return unless profile
+    if attributes['profile_ref_id'] # use the cached profile_ref_id if available
+      "ssg:rhel#{benchmark.os_major_version}|#{ShortRefId.short_ref_id(attributes['profile_ref_id'])}|#{ref_id}"
+    else # select the first canonical profile if there's a cache miss
```

Comment:
@vkrizan referring to what I mentioned in the jira task, we could theoretically just remove this part and return `nil` if the rule is not retrieved as a child of a profile. I am not exactly sure about how the remediations work here though...

Reply:
So to summarize, we need to have this context/profile aware. Returning `nil` can be okay to know what would break :slightly_smiling_face:. Using some random profile's for an unknown context (current else block) is just random, but at least it provides something. I'll leave it up to you...

Honestly, the API could have been improved to make sense like having rules as a subresource of a profile: `/v2/profiles/profileid/rules`. But with regards to what return on general all `/rules` endpoint, it's questionable whether to return each rule duplicated for each profile it is used in, or return it general unique and profile-less rules.

Reply:
This PR fixes the issue for `v1/profiles/<id>/rules`, the column will be there and the proper remediation will be returned. 

The problem is with the `/rules` endpoint, there it keeps the old and nondeterministic behavior, which I don't really like.